### PR TITLE
feat: Add PHP `extensions` handling.

### PR DESCRIPTION
**Pull request by drupol**
Hi,

Here's a small feature I just added that would be useful for me.

Basically, it infers PHP `extensions` from `composer.json`.

Let me know what you think.

### CODE REVIEW

**Comment by WyriHaximus** [comment]
@drupol Thanks for filing, loving the suggestion. But could you add the list of extensions as output to the CI workflow? https://github.com/WyriHaximus/github-action-composer-php-versions-in-range/blob/main/.github/workflows/ci.yml

**Comment by drupol** [comment]
> @drupol Thanks for filing, loving the suggestion. But could you add the list of extensions as output to the CI workflow? https://github.com/WyriHaximus/github-action-composer-php-versions-in-range/blob/main/.github/workflows/ci.yml

I'm afraid I have no idea how to do that :(

Feel free to commit in this PR if you want.

**Comment by WyriHaximus** [comment]
> I'm afraid I have no idea how to do that :(

Want me to guide you through that in this PR?

**Comment by drupol** [comment]
Of course, all the help is welcome.

File: main.js [hunk 1]
```diff
@@ -43,3 +43,32 @@ console.log(`::set-output name=version::${JSON.stringify(versions)}`);
...
+        });
+}
+
+let requiredExtensions = getExtensionsFrom('require', composerJson);
+let requiredDevExtensions = getExtensionsFrom('require-dev', composerJson);
+let allExtensions = [...requiredExtensions, ...requiredDevExtensions];
+
+console.log(`Required extensions: ${JSON.stringify(requiredExtensions)}`);
+console.log(`Required dev extensions: ${JSON.stringify(requiredDevExtensions)}`);
+console.log(`All extensions: ${JSON.stringify(allExtensions)}`);
```
Comment (WyriHaximus) [inline main.js:71]:
```suggestion
console.log(`All required extensions: ${JSON.stringify(allExtensions)}`);
```

**Review by WyriHaximus — CHANGES_REQUESTED** [review]
The code all looks good, however, there are two things missing. Most importantly an example in the readme on how to use this in a workflow.

Secondly, it's missing showing this output in the CI. My suggestion would be to add the outputs you added to this list: https://github.com/WyriHaximus/github-action-composer-php-versions-in-range/blob/main/.github/workflows/ci.yml#L16-L19

Secondly to show that, add the following at the end of the file:

```yaml
  extensions:
    name: Detected extensions on ${{ matrix.os }}
    strategy:
      matrix:
        os:
          - ubuntu-latest
          - windows-latest
          - macos-latest
    needs:
      - get-supported-php-versions
    runs-on: ${{ matrix.os }}
    steps:
      - run: |
          echo "All require extensions: ${{ needs.get-supported-php-versions.outputs.extensions }}"
          echo "Require extensions: ${{ needs.get-supported-php-versions.outputs.requiredExtensions }}"
          echo "Dev require extensions: ${{ needs.get-supported-php-versions.outputs.requiredDevExtensions }}"
```

**Review by WyriHaximus — APPROVED** [review]
LGTM, thanks :+1:!

Take your time for the readme updates, I'll merge and tag this as is. And once you have worked out a real-world example, feel free to open a PR for the readme.

**Review by WyriHaximus — APPROVED** [review]

**Comment by WyriHaximus** [comment]
Just released [`v1.7.0`](https://github.com/WyriHaximus/github-action-composer-php-versions-in-range/releases/tag/v1.7.0] with this.

**Pull request merged**